### PR TITLE
Bundle registry in npm package

### PR DESCRIPTION
**Pull request by grod220**
Related to: https://github.com/penumbra-zone/web/issues/1033

At the moment, the registry fetches the latest on github. Treating this github repo as a kind of remote service for our extension is a bit of an iffy dependency (has caused issues with schema change and recycling old jsons). This PR switches this npm package to bundle the registry and ship it together.

This means, when we update our registry, we should bump the version along with it so consumers can download the latest.

====

New addition: Adding vite to inject commit at build time so consumer can know what version of the registry it is.

The big utils file has been relocated to the /utils folder in separate files

### CODE REVIEW

File: npm/src/utils.ts
Comment (grod220) [inline npm/src/utils.ts:1]: Instead of adding to utils relocated to utils files. Should see diffs below, but they are not new.

**Comment by turbocrime** [comment]
it looks like this git commit plugin will also encounter https://github.com/penumbra-zone/web/issues/1046

File: npm/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@penumbra-labs/registry",
-  "version": "6.0.0",
+  "name": "@penumbra-zone/test-publish",
+  "version": "0.7.0",
```
Comment (turbocrime) [inline npm/package.json:3]: do these need to revert before merge?

**Comment by turbocrime** [comment]
i would configure the build to ship json files in dist, instead of bundling the json into js, so a consumer may easily access the raw json if they have some specific need.

**Review by turbocrime — APPROVED** [review]

**Comment by Valentine1898** [comment]
One disadvantage of this solution would be that when adding a new asset or channel to the registry we will have to update the extension in chrome store

File: npm/package.json (hunk 1 above)
Comment (grod220) [reply npm/package.json]: Fixed

**Comment by grod220** [comment]
> it looks like this git commit plugin will also encounter https://github.com/penumbra-zone/web/issues/1046

I suppose we are not because the action/checkout doesn't have that kind of url. Can save this for later.

> i would configure the build to ship json files in dist, instead of bundling the json into js, so a consumer may easily access the raw json if they have some specific need.

This was a fairly non-trivial thing I was playing around with for a few hours. Will open an issue about this.

> One disadvantage of this solution would be that when adding a new asset or channel to the registry we will have to update the extension in chrome store

Yes, indeed. It is trading off freshness for safety. Think it's ok for now, but we can revisit later if it becomes an issue.

**Pull request merged**